data_structures/binary_search_tree2.cpp

```diff
@@ -0,0 +1,539 @@
+/**
+ * @file binary_search_tree2.cpp
+ * @brief A generic binary search tree implementation.
+ */
+
+#include <cassert>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+/**
+ * @brief The Binary Search Tree class.
+ *
+ * @tparam T The type of the binary search tree key.
+ */
+template <class T>
+class binary_search_tree {
+ private:
+    struct bst_node {
+        T value;
```

Comment:
```suggestion
        T value; /**< 1-line description of variable */
```

Reply:
@Walt280 Thanks for adding the one line descirption but the format to add is it as suggested. 
Please update accordingly. I will "Unresolve" the older comments instead of recreating new ones. So, in this particular case, it would as such:
```suggestion
        T value; ///< The value/key of the node.
```
or 
```suggestion
        T value; /**< The value/key of the node. */
```